InnerEye/ML/lightning_models.py

```diff
@@ -412,18 +423,38 @@ def __init__(self, config: ScalarModelBase, *args, **kwargs) -> None:
         else:
             self.loss_fn = raw_loss
             self.target_indices = []
-            self.sequence_target_positions = []
+            self.sequence_target_positions = [MetricsDict.DEFAULT_HUE_KEY]
         self.is_classification_model = config.is_classification_model
         self.use_mean_teacher_model = config.compute_mean_teacher_model
         self.logits_to_posterior_fn = config.get_post_loss_logits_normalization_function()
         self.loss_type = config.loss_type
+        self.create_metrics()
+
         # TODO antonsc: Work out how we handle mean teacher model
         # if config.compute_grad_cam:
         #     model_to_evaluate = self.train_val_params.mean_teacher_model if \
         #         config.compute_mean_teacher_model else self.train_val_params.model
         #     self.guided_grad_cam = VisualizationMaps(model_to_evaluate, config)
         #     config.visualization_folder.mkdir(exist_ok=True)
 
+    def _get_metrics_classes(self) -> List[Metric]:
+        if self.is_classification_model:
+            return [Accuracy05(),
+                    AccuracyAtOptimalThreshold(),
+                    OptimalThreshold(),
+                    FalsePositiveRateOptimalThreshold(),
+                    FalseNegativeRateOptimalThreshold(),
+                    AreaUnderRocCurve(),
+                    AreaUnderPRCurve(),
+                    BinaryCrossEntropy()]
+        else:
+            return [MeanAbsoluteError(), MeanSquaredError(), ExplainedVariance()]
+
+    def create_metrics(self):
+        self.train_metrics_dict = OrderedDict(
+            [(p, self._get_metrics_classes()) for p in self.sequence_target_positions])
```

Comment:
there's a problem here I think - you are using sequence target position as the key, but later you use them as if they were hue names.

Reply:
On that note, is there a test that ensures that the loggers see the correct metric names, including prediction targets? There is a simple StoringLogger already that you could use for that.

Reply:
So I wanted to get rid of the hue name since it was just "hue + sequence_target". Here self.sequence_target_positions didn't seem to used anywhere anymore anyway, hence i "re-purposed" it. But I will rename it to `self.target_names` to avoid confusion